mesonbuild/backend/ninjabackend.py

```diff
@@ -1808,7 +1808,7 @@ def generate_genlist_for_target(self, genlist, target, outfile):
                 elem.add_item('DEPFILE', depfile)
             if len(extra_dependencies) > 0:
                 elem.add_dep(extra_dependencies)
-            elem.add_item('DESC', 'Generating $out')
+            elem.add_item('DESC', 'Generating \'%s\'.' % sole_output)
```

Comment:
Yes, that seems to be the actual problem. We need a way to signal to the escaping code that `$out` should not be escaped.